inspirehep/manage.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2015 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Manage INSPIRE instance."""
+
+from __future__ import print_function
+
+import click
+import json
+import sys
+
+from time import sleep
+
+from invenio_ext.es import es
+from invenio_ext.script import Manager
+
+manager = Manager(usage=__doc__)
+
+
+@manager.command
+def reindexhp():
+    """Reindex all Holding Pen objects."""
+    from invenio_workflows.models import BibWorkflowObject
+    from inspirehep.modules.migrator.tasks import reindex_holdingpen_object
+
+    query = BibWorkflowObject.query.with_entities(BibWorkflowObject.id).filter(
+        BibWorkflowObject.id_parent == None).all()  # noqa
+
+    click.echo(
+        click.style(
+            'Going to reindex {0} records (holdingpen)...'.format(len(query)), fg='green'
+        )
+    )
+    click.echo()
+
+    with click.progressbar(query) as holdingpen_ids:
+        for obj_id in holdingpen_ids:
+            reindex_holdingpen_object(obj_id[0])
+
+    click.echo()
+    click.echo(click.style('DONE', fg='green'))
+
+
+@manager.command
+def reindex():
+    """Reindex all records."""
+    from invenio_records.models import RecordMetadata
+    from invenio_records.tasks.index import index_record
+
+    query = RecordMetadata.query.all()
+    click.echo(
+        click.style(
+            'Going to reindex {0} records...'.format(len(query)), fg='green'
+        )
+    )
+    click.echo()
+
+    with click.progressbar(query) as record_objects:
+        for obj in record_objects:
+            index_record(obj.id, obj.json)
```

Comment:
This is going to be extremely slow because it doesn't use bulk update. However it's going to be surely correct.